src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -286,6 +318,8 @@ namespace smt::noodler {
                     if(precision == LenNodePrecision::UNDERAPPROX) {
                         ctx.get_fparams().is_underapprox = true;
                     }
+                } else {
+                    ctx.get_fparams().is_underapprox = true;
```

Comment:
Why? Shouldn't we throw error here? Or if it is unknown, then we decide we are underapproximating?

Reply:
The quantified solver is incomplete, so it can give you `l_undef`. In that case we don't want to error out, we just want to continue with a different branch - maybe we will solve it.

Reply:
Hmm, ok, it seems to make sense, but add some comment that it is there because quantified solver is incomplete.

Reply:
I have added a comment explaining why we are setting `us_underapprox = true`. Thank you for your suggestion.